### PR TITLE
:sparkles: Implement Chef InSpec Parser

### PR DESCRIPTION
**Pull request by manuel-sommer**
see issue #6990

### CODE REVIEW

**Review by mtesauro — APPROVED** [review]
Approved

File: dojo/tools/chefinspect/parser.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    def get_scan_types(self):
+        return ["Chef Inspect Log"]
+
+    def get_label_for_scan_types(self, scan_type):
+        return scan_type
+
+    def get_description_for_scan_types(self, scan_type):
+        return """Chef Inspect log file"""
+
+    def convert_score(self, raw_value):
```
Comment (cneill) [inline dojo/tools/chefinspect/parser.py:15]: It appears [from their documentation](https://docs.chef.io/inspec/profiles/controls/#control-properties) that `impact` of [0.0,1.0] is interpreted like a score from 0-100, with the following severity breakdowns.

Screenshot 2024-01-16 at 13 12 51

You can see this in practice in e.g. the [`dev-sec` MySQL baseline](https://dev-sec.io/baselines/mysql/)

Screenshot 2024-01-16 at 13 09 18

File: dojo/tools/chefinspect/parser.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                description += "zone: " + str(json_object.get('zone')) + "\n"
+                description += "office: " + str(json_object.get('office')) + "\n"
+                description += "dc: " + str(json_object.get('dc')) + "\n"
+                description += "environment: " + str(json_object.get('environment')) + "\n"
+                description += "id: " + str(json_object.get('id')) + "\n"
+                description += "control_tags: " + str(json_object.get('control_tags')) + "\n"
+                description += "platform: " + str(json_object.get('platform')) + "\n"
+                description += "profile: " + str(json_object.get('profile')) + "\n"
+                description += "group: " + str(json_object.get('group')) + "\n"
+                description += "result: " + str(json_object.get('office')) + "\n"
```
Comment (cneill) [inline dojo/tools/chefinspect/parser.py:43]: Looks like this should reference the `results` key

```suggestion
                description += "result: " + str(json_object.get('results')) + "\n"
```

**Review by cneill — CHANGES_REQUESTED** [review]
A couple notes on this one...

**Comment by manuel-sommer** [comment]
Thank you for the review @cneill. I fixed your issues.

**Review by cneill — APPROVED** [review]
Thanks! Approved

**Review by Maffooch — APPROVED** [review]

File: dojo/tools/chefinspect/parser.py (hunk 1 above)
Comment (Maffooch) [reply dojo/tools/chefinspect/parser.py:15]: @manuel-sommer I approved too quickly.. the data in the sample files provided also do not have an impact greater than 1.0, so I think the above may be the source to go with

**Comment by manuel-sommer** [comment]
I can't see your comment @Maffooch that you approve this.

File: dojo/tools/chefinspect/parser.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        if val == 0.0:
+            return "Info"
+        elif val < 4.0:
+            return "Low"
+        elif val < 7.0:
+            return "Medium"
+        elif val < 9.0:
+            return "High"
+        else:
+            return "Critical"
```
Comment (Maffooch) [inline dojo/tools/chefinspect/parser.py:26]: @manuel-sommer my apologies. I approved, and then dismissed it after seeing the feedback from @cneill was not implemented. Reasoning for the dismissed approval is in this comment
https://github.com/DefectDojo/django-DefectDojo/pull/9329#discussion_r1456069356

**Review by Maffooch — CHANGES_REQUESTED** [review]

**Comment by manuel-sommer** [comment]
Done

**Review by Maffooch — APPROVED** [review]

File: dojo/tools/chefinspect/parser.py (hunk 3 above)
Comment (cneill) [reply dojo/tools/chefinspect/parser.py]: Whoops, good catch!

**Review by grendel513 — APPROVED** [review]

**Comment by MirceaLutic** [comment]
The outputs of Chef Inspec are called "reporters". These include json format but no "logs".
How do I convert any output from inspec to the logs that dojo accepts?

**Comment by manuel-sommer** [comment]
I never used Chef Ispec Parser, just fulfilled the PR according to the linked issue description.

**Comment by mtesauro** [comment]
@Maffooch I've never used Chef Inspect either but looking at the unit test files:
https://github.com/DefectDojo/django-DefectDojo/tree/master/unittests/scans/chefinspect
They appear to be JSON files with .log extensions.

I'd suggest you compare the structure of what you're seeing to those unit test file. I may be as simple as changing the file extension :shrug:

**Pull request merged**